src/openfermion/ops/_fermion_operator.py

```diff
@@ -572,14 +576,10 @@ def __iadd__(self, addend):
         """
         if isinstance(addend, FermionOperator):
             for term in addend.terms:
-                if term in self.terms:
-                    if abs(addend.terms[term] +
-                           self.terms[term]) < EQ_TOLERANCE:
-                        del self.terms[term]
-                    else:
-                        self.terms[term] += addend.terms[term]
-                else:
-                    self.terms[term] = addend.terms[term]
+                self.terms[term] = (self.terms.get(term, 0.0) +
+                                    addend.terms[term])
+                # if abs(self.terms[term]) < EQ_TOLERANCE:
```

Comment:
What's going on here, you are commenting out lines?

Reply:
I was trying to figure out the cause of the errors with those lines included - this suggests there's something sketchy about `del self.terms[term]` causing the test failures in JelliumTest. The failures also seem to only happen in Python 2.7 which is pretty weird.

Reply:
Are you going to finish this PR or should I close it?

Reply:
I'll take a look more later this week or next - at this point it looks to be debugging something weird in the jellium module which I'm not super enthusiastic about. I might reduce the extent of the change to make life easier though